hw/ip/dma/data/dma.hjson

```diff
@@ -33,13 +33,32 @@
         { bits: "31:0", name: "PTR_ADDR", desc: "Address data pointer (word aligned) - used only in Address mode" }
       ]
     },
-    { name:     "SIZE",
-      desc:     "Number of bytes to copy - Once a value is written, the copy starts",
+    { name:     "SIZE_TR_D1",
+      desc:     "Number of bytes to copy from, defined with respect to the first dimension - Once a value is written, the copy starts",
       swaccess: "rw",
       hwaccess: "hro",
       hwqe:     "true", // enable `qe` latched signal of software write pulse
       fields: [
-        { bits: "31:0", name: "SIZE", desc: "DMA counter and start" }
+        { bits: "31:0", name: "SIZE", desc: "DMA counter D1 and start" }
+      ]
+    },
+    { name:     "SIZE_TR_D2",
+      desc:     "Number of bytes to copy from, defined with respect to the second dimension",
+      swaccess: "rw",
+      hwaccess: "hro",
+      hwqe:     "true", // enable `qe` latched signal of software write pulse
+      fields: [
+        { bits: "31:0", name: "SIZE", desc: "DMA counter D2" }
+      ]
+    },
+    { name:     "SIZE_IN",
+      desc:     "Dimensions of the input matrix",
+      swaccess: "rw",
+      hwaccess: "hro",
+      hwqe:     "true", // enable `qe` latched signal of software write pulse
+      fields: [
+        { bits: "15:0", name: "D1", desc: "Size of dimension 1" },
```

Comment:
it is not really clear to me the meaning of SIZE_IN with respect to the previous sizes - can you explain it?

Reply:
Good call... this is a register that was used in a first version of the FSM, it's totally useless, I removed it and it works.